executor/src/main/java/org/k8loud/executor/action/openstack/VerticalScalingAction.java

```diff
@@ -1,59 +1,29 @@
-import data.ExecutionRS;
 import data.Params;
-import org.jclouds.openstack.nova.v2_0.NovaApi;
-import org.jclouds.openstack.nova.v2_0.domain.Flavor;
-import org.jclouds.openstack.nova.v2_0.domain.Server;
-import org.jclouds.openstack.nova.v2_0.features.ServerApi;
-import org.k8loud.executor.action.Action;
 import org.k8loud.executor.exception.ActionException;
+import org.k8loud.executor.exception.OpenstackException;
+import org.k8loud.executor.openstack.OpenstackService;
 
-import java.util.Map;
-
-public class VerticalScalingAction extends Action {
+public class VerticalScalingAction extends OpenstackAction {
     private String region;
     private String serverId;
-    private double diskResizeValue;
-    private double ramResizeValue;
-    private double vcpusResizeValue;
-    private final NovaApi novaApi;
+    private String flavorId;
 
-    public VerticalScalingAction(Params params) throws ActionException {
-        super(params);
-        novaApi = null;
-// FIXME: 'java.lang.IllegalStateException: Unable to load cache item' when this object is instantiated in tests
-//        novaApi = OpenstackHelper.getNovaApi("openstack-nova", "demo:demo", "devstack", "http://xxx.xxx.xxx.xxx:5000/v2.0/");
+    public VerticalScalingAction(Params params, OpenstackService openstackService) throws ActionException {
+        super(params, openstackService);
     }
 
     @Override
     public void unpackParams(Params params) {
         region = params.getRequiredParam("region");
         serverId = params.getRequiredParam("serverId");
-        diskResizeValue = Double.parseDouble(params.getOptionalParam("diskResizeValue", "1"));
-        ramResizeValue = Double.parseDouble(params.getOptionalParam("ramResizeValue", "1"));
-        vcpusResizeValue = Double.parseDouble(params.getOptionalParam("vcpusResizeValue", "1"));
+        flavorId = params.getRequiredParam("flavorId");
     }
 
     @Override
-    public ExecutionRS perform() {
-        ServerApi serverApi = novaApi.getServerApi(region);
-        Server server = serverApi.get(serverId);
-
-        Flavor currentFlavor = (Flavor) server.getFlavor();
-        Flavor newFlavor = Flavor.builder()
-                .fromFlavor(currentFlavor)
-                .disk((int) (currentFlavor.getDisk() * diskResizeValue))
-                .ram((int) (currentFlavor.getRam() * ramResizeValue))
-                .vcpus((int) (currentFlavor.getVcpus() * vcpusResizeValue))
-                .build();
-
-        serverApi.resize(serverId, newFlavor.getId());
-        serverApi.confirmResize(serverId);
-        return ExecutionRS.builder()
-                .result("Success")
-                .exitCode(ExecutionExitCode.OK)
-                .build();
+    protected void performOpenstackAction() throws OpenstackException {
+        this.openstackService.resizeServer(region, serverId, flavorId);
+        //FIXME It is not working when resizing down -> maybe create snapshot, and new server with smaller flavor
```

Comment:
what is the error?

Reply:
can only resize down when flavors have same root disks sizes